bindsnet/network/topology.py

```diff
@@ -273,9 +273,17 @@ def __init__(self, source: Nodes, target: Nodes, kernel_size: Union[int, Tuple[i
 
         assert target.shape[1] == shape[1] and target.shape[2] == shape[2] and target.shape[3] == shape[3], error
 
-        self.w = kwargs.get('w', torch.rand(self.out_channels, self.in_channels, *self.kernel_size))
-        if self.wmin is not None and self.wmax is not None:
-            self.w = torch.clamp(self.w, self.wmin, self.wmax)
+        self.w = kwargs.get('w', None)
+        if self.w is None:
+            if self.wmin == -np.inf or self.wmax == np.inf:
```

Comment:
Now works the same as for `Connection`, I think the old one only did clamping, even in cases where both `wmin` and `wmax` are defined (so scaling seems more appropriate)

Reply:
Cool, thanks.